app/utility/base_planning_svc.py

```diff
@@ -80,8 +80,13 @@ async def remove_completed_links(operation, agent, links):
         """
         completed_links = [(l.command_hash if l.command_hash else l.command) for l in operation.chain
                            if l.paw == agent.paw and (l.finish or l.can_ignore())]
+
+        lat = [BasePlanningService._fil(k, 'remote.host.fqdn') for k in operation.chain if k.status != 1]
```

Comment:
Is '1' not an enum or some sort of type (w/ useful human string names) defined in c_link?

(I cant find what '1' indicates anywhere.)

Reply:
I really wish we had it enumed somewhere, but it's not. It's used to indicate that a link failed, and I had to reverse that from the web UI to figure out.

Reply:
Are these irrelevant entirely: https://github.com/mitre/caldera/blob/master/app/objects/secondclass/c_link.py#L76

Reply:
Can we update the links.states to reflect what is in the UI for consistency? After that, we can update this constant with the variable instead

Reply:
re: https://github.com/mitre/caldera/pull/1905#discussion_r501302179, they aren't irrelevant, those are just the valid status codes prior to running. Once the link runs, we get positive status codes, which don't appear to be specified anywhere... yet.

Reply:
Upon further investigation, I found where the codes are specified. They're in sandcat - https://github.com/mitre/gocat/blob/f2fbad844bf2ea8210cee9c7fa8fc2fedef62a4f/execute/execute.go#L11